protobuf-kafkaconnect-converter/src/main/java/com/amazonaws/services/schemaregistry/kafkaconnect/protobuf/toconnectschema/ProtobufSchemaToConnectSchemaConverter.java

```diff
@@ -0,0 +1,114 @@
+ * Partially inspired from https://github.com/blueapron/kafka-connect-protobuf-converter/blob/master/src/main/java/com/blueapron/connect/protobuf/ProtobufData.java#L135
+ */
+public class ProtobufSchemaToConnectSchemaConverter {
+
+    private static final Set<Descriptors.FieldDescriptor.Type> TYPES_TO_ADD_METADATA =
+        ImmutableSet.<Descriptors.FieldDescriptor.Type>builder()
+            .add(SINT32, SFIXED32, UINT32, UINT64, FIXED32, FIXED64, SFIXED64, SINT64).build();
+    private static final Integer CONVERTER_VERSION = 1;
+
+    public Schema toConnectSchema(@NonNull final Message message) {
+        final SchemaBuilder builder = SchemaBuilder.struct();
+        final Descriptors.Descriptor descriptor = message.getDescriptorForType();
+        final List<Descriptors.FieldDescriptor> fieldDescriptorList = descriptor.getFields();
+
+        builder.name(descriptor.getName());
+        builder.version(CONVERTER_VERSION);
+        builder.parameter(PROTOBUF_PACKAGE, descriptor.getFile().getPackage());
+
+        for (final Descriptors.FieldDescriptor fieldDescriptor : fieldDescriptorList) {
+            final String fieldName = fieldDescriptor.getName();
+            builder.field(fieldName, toConnectSchemaForField(fieldDescriptor));
+        }
+
+        //TODO: Add support for reading metadata from Protobuf schemas for disambiguating between INT8 and INT16
+
+        return builder.build();
+    }
+
+    private Schema toConnectSchemaForField(final Descriptors.FieldDescriptor fieldDescriptor) {
+        final Descriptors.FieldDescriptor.Type protobufType = fieldDescriptor.getType();
+        final Schema.Type connectType;
+
+        switch (protobufType) {
+            case INT32:
+            case SINT32:
+            case SFIXED32: {
+                connectType = Schema.Type.INT32;
+                break;
+            }
+            case INT64:
+            case SINT64:
+            case UINT64:
+            case FIXED64:
+            case SFIXED64:
+            case UINT32:
+            case FIXED32: {
+                connectType = Schema.Type.INT64;
+                break;
+            }
+            case FLOAT: {
+                connectType = Schema.Type.FLOAT32;
+                break;
+            }
+            case DOUBLE: {
+                connectType = Schema.Type.FLOAT64;
+                break;
+            }
+            case BOOL: {
+                connectType = Schema.Type.BOOLEAN;
+                break;
+            }
+            case STRING: {
+                connectType = Schema.Type.STRING;
+                break;
+            }
+            case BYTES: {
+                connectType = Schema.Type.BYTES;
+                break;
+            }
+            default:
+                throw new DataException("Invalid Protobuf type passed " + protobufType);
+        }
+
+        final SchemaBuilder schemaBuilder = new SchemaBuilder(connectType);
+        if (TYPES_TO_ADD_METADATA.contains(protobufType)) {
```

Comment:
What is the case when `TYPES_TO_ADD_METADATA` doesn't contain `protobufType`?

Reply:
We don't need to add it. This is only being added to distinguish between different types of INTs

Reply:
Can we add a comment here please so that it brings clarity. Also it helps the reader understand how the different int/longs are being differentiated in conversion.

Reply:
Done.